consensus/service/src/client_api_service.rs

```diff
@@ -158,3 +157,87 @@ impl<E: ConsensusEnclaveProxy, L: Ledger + Clone> ConsensusClientApi for ClientA
         });
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{client_api_service::ClientApiService, tx_manager::MockTxManager};
+    use grpcio::{ChannelBuilder, Environment, ServerBuilder};
+    use mc_attest_api::attest::Message;
+    use mc_common::{
+        logger::{test_with_logger, Logger},
+        NodeID, ResponderId,
+    };
+    use mc_consensus_api::{
+        consensus_client_grpc, consensus_client_grpc::ConsensusClientApiClient,
+        consensus_common::ProposeTxResult,
+    };
+    use mc_consensus_enclave::TxContext;
+    use mc_consensus_enclave_mock::MockConsensusEnclave;
+    use mc_transaction_core::tx::TxHash;
+    use mc_transaction_core_test_utils::{create_ledger, initialize_ledger, AccountKey};
+    use rand::{rngs::StdRng, SeedableRng};
+    use std::sync::Arc;
+
+    #[test_with_logger]
+    fn test_client_tx_propose_ok(logger: Logger) {
+        let mut consensus_enclave = MockConsensusEnclave::new();
+        consensus_enclave
+            .expect_client_tx_propose()
+            .times(1)
+            .return_const(Ok(TxContext::default()));
+
+        // Arc<dyn Fn(TxHash, Option<&NodeID>, Option<&ResponderId>) + Sync + Send>
+        let scp_client_value_sender = Arc::new(
+            |_tx_hash: TxHash, _node_id: Option<&NodeID>, _responder_id: Option<&ResponderId>| {
+                // TODO: store inputs for inspection.
+            },
+        );
+
+        // Local ledger
+        let mut ledger = create_ledger();
+        let mut rng: StdRng = SeedableRng::from_seed([62u8; 32]);
+        let sender = AccountKey::random(&mut rng);
+        let num_blocks = 5;
+        initialize_ledger(&mut ledger, num_blocks, &sender, &mut rng);
+
+        let mut tx_manager = MockTxManager::new();
+        tx_manager
+            .expect_insert()
+            .times(1)
+            .return_const(Ok(TxHash::default()));
+
+        let is_serving_fn = Arc::new(|| -> bool { true });
+
+        let client_api_service = ClientApiService::new(
+            Arc::new(consensus_enclave),
+            scp_client_value_sender,
+            ledger,
+            Arc::new(tx_manager),
+            is_serving_fn,
+            logger,
+        );
+
+        let service = consensus_client_grpc::create_consensus_client_api(client_api_service);
+
+        let env = Arc::new(Environment::new(1));
```

Comment:
Why are you doing this versus `EnvBuilder::new().build()` which is what we do everywhere else?

Reply:
Cargo-culted from https://github.com/tikv/grpc-rs/blob/master/tests-and-examples/tests/cases/health_check.rs#L54

`Environment::new` calls EnvBuilder, so this is equivalent to `EnvBuilder::new().cq_count(1).build()`. 
https://docs.rs/grpcio/0.1.0/src/grpcio/env.rs.html#105

EnvBuilder spawns `cq_count` threads for asynchronously handling requests, so maybe setting it to 1 (instead defaulting to num_cores) is a reasonable choice for a unit test?